models/utils/int_zendesk__calendar_spine.sql

```diff
@@ -8,13 +8,22 @@ with spine as (
     {% endset %}
     {% set first_date = run_query(first_date_query).columns[0][0]|string %}
     
-    {% else %} {% set first_date = "'2016-01-01'" %}
+        {% if target.type == 'postgres' %}
+            {% set first_date_adjust = "cast('" ~ first_date[0:10] ~ "' as date)" %}
+
+        {% else %}
+            {% set first_date_adjust = "'" ~ first_date[0:10] ~ "'" %}
+
+        {% endif %}
+
+    {% else %} {% set first_date_adjust = "2016-01-01" %}
```

Comment:
The second of the two conditional statements required for Postgres compatibility. The explicit casting to date was required to allow Postgres to success in downstream models. I attempted to remove the conditional and simple cast the `first_date[0:10]` as a date for all target.type's however this failed for BigQuery and Snowflake in downstream models. As such, the proposed conditional was my solution.